datus/storage/document/store.py

```
@@ -2,123 +2,372 @@
+
+        data = [chunk.to_dict() for chunk in chunks]
+
+        # Delete existing chunks with matching chunk_ids to handle deduplication,
+        # then use store_batch (table.add) which is stable in lance 0.22.0.
+        # This avoids merge_insert which has known Rust-level panics.
         self._ensure_table_ready()
-        self.create_fts_index(["chunk_text"])
+        if self.table:
+            try:
+                row_count = self.table.count_rows()
+            except Exception:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)